plugins/ltac/tacintern.ml

```diff
@@ -318,6 +318,13 @@ let evalref_of_globref ?loc ?short = function
       Nametab.pr_global_env Id.Set.empty r ++ spc () ++
       str "into an evaluable reference.")
 
+let evalref_of_globref ?loc ?short r =
+  let () = match r with
+    | GlobRef.VarRef x when (try ignore (Global.lookup_named x); false with Not_found -> true) -> ()
```

Comment:
I'm already not super fan of where clauses but this one is particularly nasty. Can you expand it into an if-then-else and optionally make this a little bit more understandable at first sight?

Reply:
changed